From b9276e5987b003b6f12370dcfae414a51b4a8201 Mon Sep 17 00:00:00 2001
From: Jonathan Dowland <jmtd@debian.org>
Date: Mon, 9 Jul 2018 23:32:05 +0100
Subject: [PATCH] tweak regex for matching SPDX-shortcode CC license

According to DEP-5, A ".0" version suffix is implied if omitted.
Therefore folks may legitimately be using e.g. CC-BY-SA-4 instead
of CC-BY-SA-4.0, and it shall be equivalent. Adjust the relevant
regexes to catch both forms.

Strictly, a suffixless CC-BY would be equivalent to CC-BY-1.0, but
that's a bit thornier and I suspect less common in practise so this
patch does not address it.
---
 tools/license-count | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/tools/license-count b/tools/license-count
index 02be6d2..63520bb 100755
--- a/tools/license-count
+++ b/tools/license-count
@@ -38,16 +38,16 @@ our @RULES = (
     [qr,(?m)^License:.*AGPL-3,                     => 'AGPL 3'],
     [qr,(?m)^License:.*Artistic(?!-),              => 'Artistic'],
     [qr,(?m)^License:.*Artistic-2,                 => 'Artistic 2.0'],
-    [qr,(?m)^License:.*CC-BY-1.0,                  => 'CC-BY 1.0'],
-    [qr,(?m)^License:.*CC-BY-SA-1.0,               => 'CC-BY-SA 1.0'],
-    [qr,(?m)^License:.*CC-BY-2.0,                  => 'CC-BY 2.0'],
-    [qr,(?m)^License:.*CC-BY-SA-2.0,               => 'CC-BY-SA 2.0'],
+    [qr,(?m)^License:.*CC-BY-1,                    => 'CC-BY 1.0'],
+    [qr,(?m)^License:.*CC-BY-SA-1,                 => 'CC-BY-SA 1.0'],
+    [qr,(?m)^License:.*CC-BY-2,                    => 'CC-BY 2.0'],
+    [qr,(?m)^License:.*CC-BY-SA-2,                 => 'CC-BY-SA 2.0'],
     [qr,(?m)^License:.*CC-BY-2.5,                  => 'CC-BY 2.5'],
     [qr,(?m)^License:.*CC-BY-SA-2.5,               => 'CC-BY-SA 2.5'],
-    [qr,(?m)^License:.*CC-BY-3.0,                  => 'CC-BY 3.0'],
-    [qr,(?m)^License:.*CC-BY-SA-3.0,               => 'CC-BY-SA 3.0'],
-    [qr,(?m)^License:.*CC-BY-4.0,                  => 'CC-BY 4.0'],
-    [qr,(?m)^License:.*CC-BY-SA-4.0,               => 'CC-BY-SA 4.0'],
+    [qr,(?m)^License:.*CC-BY-3,                    => 'CC-BY 3.0'],
+    [qr,(?m)^License:.*CC-BY-SA-3,                 => 'CC-BY-SA 3.0'],
+    [qr,(?m)^License:.*CC-BY-4,                    => 'CC-BY 4.0'],
+    [qr,(?m)^License:.*CC-BY-SA-4,                 => 'CC-BY-SA 4.0'],
     [qr,(?m)^License:.*CC0,                        => 'CC0-1.0'],
     [qr,(?m)^License:.*CDDL,                       => 'CDDL'],
     [qr,(?m)^License:.*GPL-1,                      => 'GPL 1'],
-- 
2.18.0

